src/User32.Desktop/User32.cs

```diff
@@ -57,7 +56,61 @@ public static partial class User32
         /// <summary>Windows icon or the icon of the window specified in <see cref="MENUITEMINFO.dwItemData" />.</summary>
         public static readonly IntPtr HBMMENU_SYSTEM = new IntPtr(1);
 
-        public delegate int WindowsHookDelegate(int code, IntPtr wParam, IntPtr lParam);
+        /// <summary>
+        /// An application-defined or library-defined callback function used with the <see cref="SetWindowsHookEx(WindowsHookType, IntPtr, IntPtr, int)"/> function.
+        /// This is a generic function to Hook callbacks. For specific callback functions see CallWndDelegate, CallWndRetDelegate, CBTDelegate, DebugDelegate, ForegroundIdleDelegate,
+        /// GetMsgDelegate, JournalPlaybackDelegate, JournalRecordDelegate, KeyboardDelegate, LowLevelKeyboardDelegate, MessageDelegate, MouseDelegate, LowLevelMouseDelegate,
+        /// ShellDelegate and SysMsgDelegate.
+        /// </summary>
+        /// <param name="nCode">An action code for the callback. Can be used to indicate if the hook procedure must process the message or not.</param>
+        /// <param name="wParam">First message parameter</param>
+        /// <param name="lParam">Second message parameter</param>
+        /// <returns>
+        /// An LRESULT. Usually if nCode is less than zero, the hook procedure must return the value returned by CallNextHookEx.
+        /// If nCode is greater than or equal to zero, it is highly recommended that you call CallNextHookEx and return the value it returns;
+        /// otherwise, other applications that have installed hooks will not receive hook notifications and may behave incorrectly as a result.
+        /// If the hook procedure does not call CallNextHookEx, the return value should be zero.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate int WindowsHookDelegate(int nCode, IntPtr wParam, IntPtr lParam);
+
+        /// <summary>
+        /// An application-defined callback function used with the <see cref="EnumWindowStations(WINSTAENUMPROC, IntPtr)"/> function.
+        /// </summary>
+        /// <param name="lpszWindowStation">The name of the window station.</param>
+        /// <param name="lParam">An application-defined value specified in the <see cref="EnumWindowStations(WINSTAENUMPROC, IntPtr)"/> function.</param>
+        /// <returns>To continue enumeration, the callback function must return TRUE (non-zero value). To stop enumeration, it must return FALSE (0).</returns>
+        /// <remarks>
+        /// An application must register this callback function by passing its address to <see cref="EnumWindowStations(WINSTAENUMPROC, IntPtr)"/>.
+        /// The callback function can call SetLastError to set an error code for the caller to retrieve by calling GetLastError.
+        /// </remarks>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, CharSet = CharSet.Unicode)]
+        public delegate int WINSTAENUMPROC(string lpszWindowStation, IntPtr lParam);
```

Comment:
I don't know if you remember but this and the DESKTOPENUMPROC are functions that enable the developer to return values with meaning and at the same time have the boolean semantics. I don't know any other way to map that unless we do our own BoolInt boxed type for that with specialized marshaling (@vbfox mentioned another API that behaves like this, I don't remember its name), which I would find great fun and entertainment in doing, but think might get out of this project scope.

The funny thing is that the Function documentation states this odd behavior but the callback ones don't. Really trick by the way.

Look here:

*  [EnumDesktopsFunction](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682614(v=vs.85).aspx) and [EnumWindowStationsFunction](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682644(v=vs.85).aspx):

> ## Return value
> 
> If the function succeeds, it returns the nonzero value returned by the callback function that was pointed to by `lpEnumFunc`.

*  [EnumWindowStationProc callback function](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682612(v=vs.85).aspx) and [EnumDesktopProc callback function](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682612(v=vs.85).aspx):

> ## Return value
> 
> To continue enumeration, the callback function must return TRUE. To stop enumeration, it must return FALSE.